src/gbm/gbtree.cc

```diff
@@ -254,6 +253,22 @@ void GBTree::CommitModel(std::vector<std::vector<std::unique_ptr<RegTree>>>&& ne
   GetPredictor()->UpdatePredictionCache(model_, &updaters_, num_new_trees);
 }
 
+void GBTree::Load(Json const& in) {
+  fromJson(in["gbtree_train_param"], &tparam_);
+  model_.Load(in["model"]);
+}
+
+void GBTree::Save(Json* p_out) const {
+  auto& out = *p_out;
+  out["name"] = String("gbtree");
+  out["gbtree_train_param"] = toJson(tparam_);
```

Comment:
I do not remember whether our index operation takes temp objects on the left side (&&). Does it?

Reply:
@trams It's a long way:

`gbtree_train_param` is not in the map, so `std::map::operator[]` initializes one for us, returns it as an reference of `Json` object all the way back to here,  then the move assignment operator is invoked ...

Reply:
Will submit an optimization PR for JSON.  Right now I just want to get parameter IO sorted out for distributed settings.  So that I don't have to fix any more bugs for it in the next major release. ;-)